### DISCUSSION TITLE
installation permission errors

### DISCUSSION THREAD

**Discussion by LoupiBe** [ericaltendorf/plotman#593] [Q&A] [answered]
Hello there,

running Ubuntu Serer 20.04 LTS, latest chia installed (1.1.6 dev0), I tried installing plotman by following procedure, i.e.:

`Collecting git+https://github.com/ericaltendorf/plotman@development
 Cloning https://github.com/ericaltendorf/plotman (to revision development) to /tmp/pip-req-build-l5zmslat
Running command git clone -q https://github.com/ericaltendorf/plotman /tmp/pip-req-build-l5zmslat
Running command git checkout -b development --track origin/development

Switched to a new branch 'development'

Branch 'development' set up to track remote branch 'development' from 'origin'.
Installing build dependencies ... done
Getting requirements to build wheel ... done
Preparing wheel metadata ... done
      Using cached PyYAML-5.4.1-cp38-cp38-manylinux1_x86_64.whl (662 kB)
Collecting marshmallow~=3.12
      Using cached marshmallow-3.12.1-py2.py3-none-any.whl (47 kB)
Collecting appdirs~=1.4
      Using cached appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting desert==2020.11.18
      Using cached desert-2020.11.18-py3-none-any.whl (9.5 kB)
Collecting pendulum~=2.1
      Using cached pendulum-2.1.2-cp38-cp38-manylinux1_x86_64.whl (155 kB)
Collecting psutil~=5.8
      Using cached psutil-5.8.0-cp38-cp38-manylinux2010_x86_64.whl (296 kB)
Collecting texttable~=1.6
      Using cached texttable-1.6.3-py2.py3-none-any.whl (10 kB)
Collecting attrs==21.2
      Using cached attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting click~=7.1
      Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Collecting typing-inspect
      Using cached typing_inspect-0.6.0-py3-none-any.whl (8.1 kB)
Collecting python-dateutil<3.0,>=2.6
      Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting pytzdata>=2020.1
      Using cached pytzdata-2020.1-py2.py3-none-any.whl (489 kB)
Collecting six>=1.5
      Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting mypy-extensions>=0.3.0
      Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting typing-extensions>=3.7.4
      Using cached typing_extensions-3.10.0.0-py3-none-any.whl (26 kB)
Building wheels for collected packages: plotman
	Building wheel for plotman (PEP 517) ... done
	Created wheel for plotman: filename=plotman-0.3.1+dev-py3-none-any.whl size=58288 sha256=5cedab7134d64abc72ec531491317e6136257bfe402cbe9c52cbec0c054916c1
	Stored in directory: /tmp/pip-ephem-wheel-cache-j0bbh51q/wheels/22/30/10/eefae3a81b37cabff2ed6e11eaafc3c5f69313d66b7d7f4456
Successfully built plotman
Installing collected packages: typing-extensions, six, mypy-extensions, typing-inspect, pytzdata, python-dateutil, marshmallow, attrs, texttable, pyyaml, psutil, pendulum, desert, click, appdirs, plotman
	Attempting uninstall: typing-extensions
		Found existing installation: typing-extensions 3.7.4.3
		Uninstalling typing-extensions-3.7.4.3:
ERROR: Could not install packages due to an OSError: [Errno 13] Permission denied: 'INSTALLER'
Check the permissions.    `

I do not know which permissions to set up to make the installation complete.

I also tried to install the @main and @development branch without success.

Could someone have a clue what's going on?
Thank you in advance!

**Comment by calarndt** [comment, marked as answer]
don't install plotman that way. in my experience you'll never get a working installation.
Instead install pip
sudo apt install python3-pip
then head over to the plotman page at pypi.org
https://pypi.org/project/plotman/
and grab this command:
`pip` install --force-reinstall `git+https://github.com/ericaltendorf/plotman@main`
doing that will get you a working plotman every time.
be aware you must add the PATH it recommends
and you must . ./activate your chia environment

**Reply by LoupiBe** [reply]
Thanks! did the trick indeed!
